src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs

```diff
@@ -237,6 +237,18 @@ public PingReply Send(IPAddress address, int timeout, byte[] buffer, PingOptions
             }
         }
 
+        public PingReply Send(IPAddress address, TimeSpan timeout, byte[]? buffer = null, PingOptions? options = null)
+        {
+            int milliseconds = (int)timeout.TotalMilliseconds;
```

Comment:
for all these casts, we have to decide whether to add `checked`. We probably should.

Also anything like these can be written as an expression bodied one liner.